### PR TITLE
tools: safechroot: TestInitializeShouldCreateChroot: fix if condition

### PR DESCRIPTION
**Pull request by mfrw**
Signed-off-by: Muhammad Falak R Wani <falakreyaz@gmail.com>

###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [x] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/tools/cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [x] Ready to merge

---

###### Summary 

What does the PR accomplish, why was it needed?
- Closes: #3244

###### Change Log  

- Fix if/else condition in `TestInitializeShouldCreateRoot`

###### Does this affect the toolchain?  

**YES**

###### Associated issues  

- #3244

###### Links to CVEs  
- NA

###### Test Methodology

- Local Build.
- Buddy Build: [223399](https://dev.azure.com/mariner-org/mariner/_build/results?buildId=223399&view=results)

### CODE REVIEW

File: toolkit/tools/internal/safechroot/safechroot_test.go [hunk 1]
```diff
@@ -71,7 +71,7 @@ func TestInitializeShouldCreateRoot(t *testing.T) {
 	if buildpipeline.IsRegularBuild() {
 		assert.True(t, !os.IsNotExist(err))
 	} else {
-		assert.True(t, !os.IsNotExist(err))
+		assert.True(t, os.IsNotExist(err))
```
Comment (kanikanema) [inline toolkit/tools/internal/safechroot/safechroot_test.go:74]: Looks like we expect err=1 for non-regular build which implies we support only regular builds. Should we assert IsRegularBuild as well?

**Review by PawelWMS — APPROVED** [review]

File: toolkit/tools/internal/safechroot/safechroot_test.go (hunk 1 above)
Comment (mfrw) [reply toolkit/tools/internal/safechroot/safechroot_test.go:74]: @kanikanema:
In my opinion `err` is a type; and internally calls [underlyingError](https://cs.opensource.google/go/go/+/refs/tags/go1.19:src/os/error.go;l=133;drc=0f6ee42fe063a48d7825bc03097bbb714aafdb7d;bpv=0;bpt=1).
In anycase, could you elaborate a little more;

Comment (mfrw) [reply toolkit/tools/internal/safechroot/safechroot_test.go]: @kanikanema on further inspection; it seems the original `if` condition was reduandant.

Buddy Build: [223399](https://dev.azure.com/mariner-org/mariner/_build/results?buildId=223399&view=results)

**Review by kanikanema — APPROVED** [review]
Looks good. Thanks for addressing the comment.

**Pull request merged**